archive/progress_common/src/main/scala/uk/ac/wellcome/platform/archive/common/progress/monitor/ProgressTracker.scala

```diff
@@ -84,27 +86,33 @@ class ProgressTracker @Inject()(
       .given(attributeExists('id))
       .update('id -> update.id, mergedUpdate)
 
-    Scanamo.exec(dynamoClient)(ops) match {
+    Scanamo.exec(dynamoDbClient)(ops) match {
       case Left(ConditionNotMet(e: ConditionalCheckFailedException)) => {
         val idConstraintError =
           IdConstraintError(s"Progress does not exist for id:${update.id}", e)
 
         Failure(idConstraintError)
       }
-
       case Left(scanamoError) => {
         val exception = new RuntimeException(scanamoError.toString)
         warn(s"Failed to update Dynamo record: ${update.id}", exception)
         Failure(exception)
       }
-
-      case r @ Right(progress) => {
+      case Right(progress) => {
         debug(
           s"Successfully updated Dynamo record: ${update.id}, got $progress")
         Success(progress)
       }
     }
   }
+
+  def findByBagId(bagId: BagId) = {
+    val query = Table[BagIngest](dynamoConfig.table)
+      .index(dynamoConfig.index)
+      .limit(30)
```

Comment:
What’s the rationale behind the limit?

Reply:
To keep this a simple change and eliminate pagination and there isn't really a sensible use case at the moment for returning many ingests.  Maybe a comment is needed?

Reply:
I think as this is not going to be part of the documented API it should probably have some comments explaining that anyway.